tools/src/main/java/org/bitcoinj/tools/BlockImporter.java

```diff
@@ -42,21 +42,26 @@ public static void main(String[] args) throws BlockStoreException, VerificationE
         new Context(params);
         
         BlockStore store;
-        if (args[1].equals("H2")) {
```

Comment:
This particular refactoring I'm not very fond of, because of the ugly switch syntax. Of course the repeated use of `args[1]` is ugly, but this could also be assigned to a temporary value. What do you think? In general, do you prefer `switch` to `if`?

Reply:
Btw. just a nitpick, but it's also not a JDK7 or JDK8 upgrade. Switch has been available from Java day one afair.

Reply:
I think the switch syntax (despite the ugly way `break` is required) is much more readable. It's clear that only one of the `case` s ever fires and it's really easy to scan and find the `case` that you're looking for.

Reply:
Also note that further improvements to `switch` are coming in newer JDKs.